src/components/TodoList/TodoList.tsx

```diff
@@ -1,222 +1,129 @@
 /* eslint-disable */
-import React from 'react';
+import React, { useEffect, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { useSelector } from 'react-redux';
+import { RootState } from '../../app/store';
+import classNames from 'classnames';
+import { Todo } from '../../types/Todo';
+import { actions } from '../../features/todos';
+import { actions as currentTodoActions } from '../../features/currentTodo';
+import { getTodos } from '../../api';
 
 export const TodoList: React.FC = () => {
-  return (
-    <>
-      <p className="notification is-warning">
-        There are no todos matching current filter criteria
-      </p>
+  const dispatch = useDispatch();
+  const todos: Todo[] = useSelector((state: RootState) => state.todos);
 
-      <table className="table is-narrow is-fullwidth">
-        <thead>
-          <tr>
-            <th>#</th>
+  const status = useSelector<RootState>(state => state.filter.status);
+  const text = useSelector<RootState, string>(state => state.filter.text);
+  const { id: currentId } =
+    useSelector((state: RootState) => state.currentTodo) || {};
 
-            <th>
-              <span className="icon">
-                <i className="fas fa-check" />
-              </span>
-            </th>
+  const visibleTodos = useMemo(() => {
+    let newTodos = [...todos];
 
-            <th>Title</th>
-            <th> </th>
-          </tr>
-        </thead>
+    if (status === 'all') {
+      newTodos = todos;
+    }
 
-        <tbody>
-          <tr data-cy="todo">
-            <td className="is-vcentered">1</td>
-            <td className="is-vcentered"> </td>
+    if (status === 'active') {
+      newTodos = todos.filter(todo => !todo.completed);
+    }
 
-            <td className="is-vcentered is-expanded">
-              <p className="has-text-danger">delectus aut autem</p>
-            </td>
+    if (status === 'completed') {
+      newTodos = todos.filter(todo => todo.completed);
+    }
 
```

Comment:
Use switch case instead of 3 if for better readable, and create a enum for words 'all', 'completed', 'active'